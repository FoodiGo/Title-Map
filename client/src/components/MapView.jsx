import React from 'react';
import PropTypes from 'prop-types';
import {
  Card, CardImg,
  CardBody, CardTitle, CardSubtitle,
} from 'reactstrap';
import IoIosLocation from 'react-icons/lib/io/ios-location';

const MapView = ({ map }) => (
  <div>
    <Card>
      <CardImg top width="100%" src={map.image} alt="Map image" />
      <CardBody>
        <div className="cardTitle">
          <IoIosLocation />
          <CardTitle>{map.address}</CardTitle>
        </div>
        <CardSubtitle>{map.phonenumber}</CardSubtitle>
        {/* <link href={`https://maps.google.com/?q=${map.address}`}>Get Directions</link> */}
      </CardBody>
    </Card>
  </div>
);

MapView.propTypes = {
  map: PropTypes.shape({
    id: PropTypes.number,
    address: PropTypes.string,
    image: PropTypes.string,
    phonenumber: PropTypes.string,
    id_restaurant: PropTypes.number,
  }).isRequired,
};

export default MapView;
