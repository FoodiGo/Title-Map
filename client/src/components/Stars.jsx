import PropTypes from 'prop-types';
import React from 'react';
import Star from './Star';

const uniqid = require('uniqid');

const Stars = ({ rating, reviews }) => {
  const starNum = [];
  for (let i = 0; i < 5; i += 1) {
    i < rating ? starNum.push(<Star className="redStar star" key={uniqid()} />) :
      starNum.push(<Star className="greyStar star" key={uniqid()} />);
  }
  return (
    <div>
      <div className="starContainer" >
        { starNum }
        <div className="reviewText">{`${reviews} reviews`}</div>
      </div>
    </div>
  );
};

Stars.propTypes = {
  rating: PropTypes.number.isRequired,
  reviews: PropTypes.number.isRequired,
};

module.exports = Stars;
