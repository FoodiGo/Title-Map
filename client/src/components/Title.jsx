import React from 'react';
import PropTypes from 'prop-types';
import Stars from './Stars';

const Title = ({ title, reviews }) => (
  <div>
    <div className="title">{title.title}</div>
    <Stars rating={title.rating} reviews={reviews} />
  </div>
);

Title.propTypes = {
  title: PropTypes.shape({
    rating: PropTypes.number,
    price: PropTypes.string,
    id: PropTypes.number,
    title: PropTypes.string,
  }).isRequired,
  reviews: PropTypes.number.isRequired,
};

module.exports = Title;
