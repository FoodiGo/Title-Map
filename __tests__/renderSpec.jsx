import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import MapView from '../client/src/components/MapView';
import Title from '../client/src/components/Title';
import Stars from '../client/src/components/Stars';
import App from '../client/src/index';

configure({ adapter: new Adapter() });

describe('App render', () => {
  test('App component should be stateful', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state('title')).toHaveProperty('title');
  });
});

describe('MapComponent render', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<MapView map={
            {
            id: 30,
            address: '51052 Roob Forge West Chesley Utah, 11729-3188',
            image: 'http://lorempixel.com/640/480',
            phonenumber: '942-284-4906',
            id_restaurant: 130,
            }
          }
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Title Component render', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<Title
        title={
            {
              id: 130,
              title: 'Nienow, Bradtke and Hills',
              numStars: 4,
              price: '$$$$',
            }
          }
        types={['Indian']}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Stars Component render', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<Stars numstars={4} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
