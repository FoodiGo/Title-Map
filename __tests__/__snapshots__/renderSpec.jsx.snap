// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent render it renders correctly 1`] = `
<div>
  <div
    className="card"
  >
    <img
      alt="Map image"
      className="card-img-top"
      src="http://lorempixel.com/640/480"
      width="100%"
    />
    <div
      className="card-body"
    >
      <ul>
        <li>
          <svg
            className="locationIcon icon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m19.8 2.5c6.2 0 11.2 5 11.2 11.3 0 8.7-11.2 23.7-11.2 23.7s-11.3-15-11.3-23.7c0-6.3 5-11.3 11.3-11.3z m0 14.9c2 0 3.6-1.6 3.6-3.6s-1.6-3.7-3.6-3.7-3.7 1.6-3.7 3.7 1.6 3.6 3.7 3.6z"
              />
            </g>
          </svg>
          <span
            className="address cardText"
          >
            51052 Roob Forge West Chesley Utah, 11729-3188
          </span>
          <svg
            className="editIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m29.8 14.8l-19.6 19.6-7.1-7.1 19.6-19.6z m4.6-7.9c0.9 0.9 0.8 2.5-0.2 3.5l-3.5 3.5-7.1-7.1 3.5-3.5c1-1 2.6-1.1 3.5-0.2z m-31.9 21.5l6.6 6.6-9.1 2.5z"
              />
            </g>
          </svg>
          <a
            className="edit cardText"
            href="#"
          >
            Edit
          </a>
           
        </li>
        <li>
          <svg
            className="icon phoneIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m32.3 26.3c1.2 1.2 2 2.7 0 5.1-2 2.4-3.3 2.4-5 2.4-1.9 0-4.6-1.1-7.3-3.2-2.5-1.8-3.6-2.6-6.1-5.2-2.7-2.7-4.3-5-6.2-8.4-2.1-3.9-1.9-5.7-1.4-7.2 0.3-0.9 1.1-1.8 2.2-2.5l0.1-0.1c0.4-0.3 1.4-0.9 2.5-0.9 1 0 1.9 0.4 2.6 1.4l0.1 0c0.9 1.1 1.9 2.5 2.5 3.7 0.8 1.6 0.8 3-0.1 4.1-0.7 1-0.9 1.5-0.9 1.8s0.4 0.7 1 1.5l0.1 0c1.2 1.4 1.6 1.8 2 2.3 0.1 0.1 0.3 0.2 0.3 0.2l0.2 0.2c0.5 0.5 0.8 0.9 2.3 2.2l0 0.1c0.6 0.4 0.7 0.6 1 0.6 0.4 0 1.2-0.6 2.2-1.2 0.4-0.2 0.9-0.4 1.4-0.4 2.1 0 5.3 2.3 6.4 3.5z"
              />
            </g>
          </svg>
          <span
            className="phoneNumber cardText"
          >
            942-284-4906
          </span>
        </li>
        <li>
          <svg
            className="icon mapIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m17.5 2.5c9.7 0 17.5 7.8 17.5 17.5s-7.8 17.5-17.5 17.5-17.5-7.8-17.5-17.5c0-3 0.8-5.9 2.1-8.4-1.2-1-2.1-2.4-2.1-4.1 0-2.7 2.3-5 5-5 1.7 0 3.1 0.9 4.1 2.1 2.5-1.3 5.4-2.1 8.4-2.1z m-15 5c0 0.8 0.4 1.6 1 2 1-1.3 2.2-2.5 3.5-3.5-0.4-0.6-1.2-1-2-1-1.4 0-2.5 1.1-2.5 2.5z m25.6 23.1c2.8-2.8 4.4-6.6 4.4-10.6s-1.6-7.8-4.4-10.6-6.6-4.4-10.6-4.4-7.8 1.6-10.6 4.4-4.4 6.6-4.4 10.6 1.6 7.8 4.4 10.6 6.6 4.4 10.6 4.4 7.8-1.6 10.6-4.4z m-3.1-20.6s-4.2 10.2-5.6 11.9-9.4 8.1-9.4 8.1 4.2-10.3 5.6-11.9 9.4-8.1 9.4-8.1z"
              />
            </g>
          </svg>
          <a
            className="directions cardText"
            href="https://maps.google.com/?q=51052 Roob Forge West Chesley Utah, 11729-3188"
          >
            Get Directions
          </a>
           
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Stars Component render it renders correctly 1`] = `
<div>
  <div
    className="starContainer"
  >
    <div
      className="greyStar star"
    >
      <img
        alt="star img"
        className="starImage"
        src="test-file-stub"
      />
    </div>
    <div
      className="greyStar star"
    >
      <img
        alt="star img"
        className="starImage"
        src="test-file-stub"
      />
    </div>
    <div
      className="greyStar star"
    >
      <img
        alt="star img"
        className="starImage"
        src="test-file-stub"
      />
    </div>
    <div
      className="greyStar star"
    >
      <img
        alt="star img"
        className="starImage"
        src="test-file-stub"
      />
    </div>
    <div
      className="greyStar star"
    >
      <img
        alt="star img"
        className="starImage"
        src="test-file-stub"
      />
    </div>
    <div
      className="reviewText"
    >
      undefined reviews
    </div>
  </div>
</div>
`;

exports[`Title Component render it renders correctly 1`] = `
<div>
  <div
    className="title"
  >
    Nienow, Bradtke and Hills
  </div>
  <div>
    <div
      className="starContainer"
    >
      <div
        className="greyStar star"
      >
        <img
          alt="star img"
          className="starImage"
          src="test-file-stub"
        />
      </div>
      <div
        className="greyStar star"
      >
        <img
          alt="star img"
          className="starImage"
          src="test-file-stub"
        />
      </div>
      <div
        className="greyStar star"
      >
        <img
          alt="star img"
          className="starImage"
          src="test-file-stub"
        />
      </div>
      <div
        className="greyStar star"
      >
        <img
          alt="star img"
          className="starImage"
          src="test-file-stub"
        />
      </div>
      <div
        className="greyStar star"
      >
        <img
          alt="star img"
          className="starImage"
          src="test-file-stub"
        />
      </div>
      <div
        className="reviewText"
      >
        undefined reviews
      </div>
    </div>
  </div>
</div>
`;
